migration/incremental/021-post-restore.sql

```diff
@@ -0,0 +1,72 @@
+
+CREATE FUNCTION public.promscale_post_restore()
+RETURNS void
+SECURITY DEFINER
+SET search_path = pg_catalog
+AS $func$
+DECLARE
+    _sql text;
+BEGIN
+    CREATE TRIGGER ts_insert_blocker
+    BEFORE INSERT ON _ps_trace.span
+    FOR EACH ROW EXECUTE FUNCTION _timescaledb_internal.insert_blocker();
+
+    CREATE TRIGGER ts_insert_blocker
+    BEFORE INSERT ON _ps_trace.event
+    FOR EACH ROW EXECUTE FUNCTION _timescaledb_internal.insert_blocker();
+
+    CREATE TRIGGER ts_insert_blocker
+    BEFORE INSERT ON _ps_trace.link
+    FOR EACH ROW EXECUTE FUNCTION _timescaledb_internal.insert_blocker();
+
+    FOR _sql IN
+    (
+        SELECT format('DROP TRIGGER IF EXISTS ts_insert_blocker ON %I.%I RESTRICT', k.schema_name, k.table_name)
+        FROM _timescaledb_catalog.hypertable h
+        INNER JOIN _timescaledb_catalog.chunk k ON (k.hypertable_id = h.id)
+        WHERE h.schema_name = '_ps_trace'
+        AND h.table_name IN ('span', 'link', 'event')
+        ORDER BY h.table_name, k.table_name
+    )
+    LOOP
+        EXECUTE _sql;
+    END LOOP
+    ;
+
+    SELECT format('ALTER TABLE _ps_trace.tag_key ALTER COLUMN id RESTART WITH %s', max(id) + 1)
```

Comment:
why do we need all these?

Reply:
also do these queries use indexes correctly to find the max?

Reply:
pg_dump doesn't appear to try to update sequences for extension config tables

Reply:
ohhhh wait!

```
The pg_extension_config_dump() function can also register sequences, so 
that the current properties of the registered sequences are saved by 
pg_dump and later restored. The function can support sequences either 
explicitely created with an ALTER SEQUENCE statement or implicitely 
created when a table contains SERIAL or BIGSERIAL columns. Note that the 
sequences associated to SERIAL or BIGSERIAL columns of a configuration 
table need to be registered using the pg_extension_config_dump() 
function if one wants to restore the properties they had at pg_dump time
```
https://www.postgresql.org/message-id/568C27BD.8030402%40free.fr

let me try that

Reply:
jeez that's so much easier. why did that not occur to me earlier?!?

resolved in [486d8b5](https://github.com/timescale/promscale_extension/pull/185/commits/486d8b5e16f713ac139bbee875c8c801f1d9868c)